src/wp-includes/class-wp-query.php

```diff
@@ -3571,7 +3571,13 @@ public function the_post() {
 		global $post;
 
 		if ( ! $this->in_the_loop ) {
-			update_post_author_caches( $this->posts );
+			$post_ids = array_filter( $this->posts, 'is_numeric' );
+			$post_ids = array_filter( $post_ids );
```

Comment:
```suggestion
			$post_ids = array_filter( array_filter( $this->posts, 'is_numeric' ) );
```

We could do this, might be a little cleaner.

Reply:
I'd pushed 3b5fca16f61d6f497bab2965bd2c9a6d8e0e5459 before seeing this, on the basis that comments are needed to explain why array_filter is called twice.